lib/clearance/back_door.rb

```diff
@@ -49,7 +49,8 @@ def call(env)
     # @api private
     def sign_in_through_the_back_door(env)
       params = Rack::Utils.parse_query(env["QUERY_STRING"])
-      user_param = params["as"]
+      user_param = params.delete("as")
```

Comment:
If someone had an `as` param in a test, but was not using it for purposes of the clearance backdoor, would this have the unintentional side effect of the removing their param?

Reply:
Even before this change, passing an `as` parameter for other uses would have strange side effects. We'd be trying to use that param as a user ID before passing it through to the app, so it would either be inadvertently logging in as some random user if it happened to match a valid user ID, or it would be raising a record not found exception.

I guess if someone was passing a block that looks up the user in a way that doesn't raise (e.g. `config.middleware.use Clearance::BackDoor { |id| User.find_by(id: id) }`) then it wouldn't raise, but it would still unintentionally log them in.

Maybe it's worth allowing users to specify what this param is called to avoid collisions, but that's probably out-of-scope for this PR.